cmd/influx/main.go

```diff
@@ -185,26 +186,72 @@ func newApiClient(ctx *cli.Context, cli *internal.CLI, injectToken bool) (*api.A
 	return api.NewAPIClient(apiConfig), nil
 }
 
-// standardCtx returns a context that will cancel on SIGINT and SIGTERM.
-func standardCtx(ctx *cli.Context) context.Context {
-	return signals.WithStandardSignals(ctx.Context)
-}
-
 var app = cli.App{
 	Name:      "influx",
 	Usage:     "Influx Client",
 	UsageText: "influx [command]",
 	Commands: []*cli.Command{
 		&versionCmd,
 		&pingCmd,
-		&setupCmd,
-		&writeCmd,
-		&bucketCmd,
+		newSetupCmd(),
+		newWriteCmd(),
+		newBucketCmd(),
 	},
 }
 
+func withCli() cli.BeforeFunc {
+	return func(ctx *cli.Context) error {
+		c, err := newCli(ctx)
+		if err != nil {
+			return err
+		}
+		ctx.App.Metadata["cli"] = c
+		return nil
+	}
+}
+
+func getCLI(ctx *cli.Context) *internal.CLI {
+	i, ok := ctx.App.Metadata["cli"].(*internal.CLI)
+	if !ok {
+		panic("missing CLI")
+	}
+	return i
+}
+
+func withApi() cli.BeforeFunc {
```

Comment:
Could we have this take a `bool` arg and only build/register one client, so we can drop the `getAPINoToken` function? Or is the extra explicitness of `getAPI` vs. `getAPINoToken` worth it?

Reply:
I originally had it that way - and then thought it looked clearer at the call site that you were requesting the API sans token. Don’t feel very strongly about either approach

Reply:
I also like the explicitness at the call site, the more that I look at it. It's probably fine in practice, but bothers me to construct 2 clients when you're guaranteed to only use 1.

Assuming we keep both `getAPI` and `getAPINoToken`, what do you think between these options?
1. Have `withApi` take a bool arg that determines whether or not a token is injected
   * The key used in `ctx.App.Metadata` is also determined by the bool
   * Users need to call the correct `get` function based on the bool they passed, otherwise they'll see a panic
2. Have separate `withApi` and `withApiNoToken` middlewares
   * Could be `var`s that call an extracted `makeFn`, to avoid duplication
   * Users need to call the correct `get` function based on the middleware they used, otherwise they'll see a panic
3. Leave things as-is, even though we're building a redundant client it's worth reducing the chances of a panic when devs copy-paste-adjust for new commands

Reply:
Given how light weight the client is, I was't concerned about building a second one, however, I can just move back to the `withApi(bool)` and move some of the burden back to the developer to define the correct version for their CLI command.

Reply:
I went ahead and pushed the change to unblock merge